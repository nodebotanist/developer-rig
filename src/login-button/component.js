import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

export class LoginButton extends Component {
  render() {
    const rigAuthUrl = 'https://id.twitch.tv/oauth2/authorize?client_id=2dhk7ewe0el4fyu2y8af42v3i0znx5&redirect_uri=https://localhost.rig.twitch.tv:3000&response_type=token&scope=user:edit+extensions:edit:catalog';
    return (
      <div className="login-button">
        <a href={rigAuthUrl}>
          <button
          // onClick={this.props.loginHandler}>
          >Log In</button>
        </a>
      </div>
    );
  }
}

LoginButton.propTypes = {
  loginHandler: PropTypes.func.isRequired
};
