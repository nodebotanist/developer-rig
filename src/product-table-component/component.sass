.product-table
  min-width: 1000px
  max-width: 1250px
  margin-top: 3rem
  overflow-x: hidden

.product-table__error
  background: #faf9fa
  border: 1px solid #dad8de
  border-left-width: 0.3rem
  border-left-color: #fc3636
  padding: 1rem
  margin-bottom: 1rem

  *
    margin: 0
    padding: 0
    box-sizing: border-box
    line-height: 2rem

  h4
    font-weight: 600
    font-size: 1.2rem

.product-table__header
  display: flex
  justify-content: space-between
  align-items: center
  margin-bottom: 3px
  font-family: HelveticaNeue-Medium;

  *
    padding-left: 3px

  .text
    width: 195.5px

  .select
    width: 162px

.product-table__row
  display: flex
  justify-content: space-between
  align-items: center
  margin-top: 10px
  font-family: HelveticaNeue
  
  .form__input
    box-shadow: inset 0 0 0 1px #dad8de, 0 0 0 transparent
    background-color: white
    color: #433f4a
    width: 175.5px
    height: 20px
    padding: 5px 10px
    border: none
    border-radius: 0
    font-size: 14px
    line-height: 14px
    -webkit-appearance: none
    -moz-appearance: none
    appearance: none


  select.form__input
    width: 162px
    height: 30px
    cursor: pointer
    background-image: url("data:image/svg+xml,%3Csvg%20viewBox%3D%220%200%2012%2012%22%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20fill%3D%22%23898395%22%3E%3Cpolygon%20points%3D%226%2C1%2010%2C5%202%2C5%22/%3E%20%3Cpolygon%20points%3D%226%2C11%2010%2C7%202%2C7%22/%3E%3C/svg%3E")
    background-repeat: no-repeat
    background-position: right 10px center
    background-size: 14px

.product-table__add-btn
  background: #6441a4
  border: none
  color: white
  display: inline-block
  position: relative
  float: right
  width: 103px
  margin-top: 20px
  padding: 0 1rem
  transition: background 0.12s ease-in, color 0.12s ease-in, box-shadow 0.12s ease-in
  font-size: 12px
  font-weight: 400
  line-height: 2rem
  white-space: nowrap
  cursor: pointer