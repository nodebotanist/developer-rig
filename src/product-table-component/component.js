import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';
import { ProductRow } from './product-row';
import { fetchProducts, saveProduct } from '../util/api';

export class ProductTable extends Component {
  constructor(props) {
    super(props);

    this.state = {
      products: [{
        sku: '',
        amount: 1,
        inDevelopment: 'true',
        displayName: '',
        broadcast: 'true'
      }],
      error: ''
    };
  }

  componentDidMount() {
    fetchProducts(
      'api.twitch.tv',
      this.props.clientId,
      this._handleFetchProductsSuccess.bind(this),
      this._handleFetchProductsError.bind(this)
    );
  }

  handleValueChange(index, event) {
    const value = event.target.value;
    const fieldName = event.target.name;
    const partial = {
      [fieldName]: value,
      dirty: true
    };
    this._updateProduct(index, partial);
  }

  handleAddProductClick(event) {
    this.setState(prevState => {
      const products = [...prevState['products']];
      const product = {
        sku: '',
        amount: 1,
        inDevelopment: 'true',
        displayName: '',
        broadcast: 'true',
        dirty: true
      };
      products.push(product)
      return { products: products };
    });
  }

  handleSaveProductsClick(event) {
    const dirtyProducts = this.state.products.map((p, i) => {
      if (p.dirty) {
        p.saving = true
        saveProduct(
          'api.twitch.tv',
          this.props.clientId,
          this.props.token,
          p,
          this._handleSaveProductSuccess.bind(this, i),
          this._handleSaveProductError.bind(this, i)
        );
      }
      return p;
    });
    this.setState({
      products: dirtyProducts
    });
  }

  render() {
    const productRows = this.state.products.map((p, i) => {
      return (
        <ProductRow key={i} product={p} handleValueChange={this.handleValueChange.bind(this, i)} />
      );
    });

    return (
      <div className="product-table">
        {this.state.error &&
          <div className="product-table__error">
            <h4>Error getting products.</h4>
            <p>{this.state.error}</p>
          </div>
        }
        <div className="product-table__header">
          <div className="text-col">Product Name</div>
          <div className="text-col">SKU</div>
          <div className="text-col">Amount (in Bits)</div>
          <div className="select-col">In Development</div>
          <div className="select-col">Broadcast</div>
          <div className="dirty-col"></div>
        </div>
        {productRows}
        <div className="product-table__buttons">
          <button className="product-table__add-button" onClick={this.handleAddProductClick.bind(this)}>
            Add Product
          </button>
          <button className="product-table__save-button" onClick={this.handleSaveProductsClick.bind(this)}>
            Save All
          </button>
        </div>
      </div>
    );
  }

  _updateProduct(index, partial) {
    this.setState(prevState => {
      const products = prevState.products.map((product, idx) => {
        if (idx === index) {
           return {
              ...product,
              ...partial
           };
        }
        return product;
     });
     return { products: products };
    });
  }

  _handleFetchProductsSuccess(products) {
    this.setState({ products: products });
  }

  _handleFetchProductsError(error) {
    this.setState({ error: error });
  }

  _handleSaveProductSuccess(index) {
    const partial = {
      dirty: false
    };
    this._updateProduct(index, partial);
  }

  _handleSaveProductError(index, error) {
    const partial = {
      error: error
    };
    this._updateProduct(index, partial);
  }
}

ProductTable.propTypes = {
  clientId: PropTypes.string.isRequired,
  token: PropTypes.string.isRequired
}