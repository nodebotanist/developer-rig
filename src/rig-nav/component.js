import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';
import { EXTENSION_VIEWS, BROADCASTER_CONFIG, LIVE_CONFIG, CONFIGURATIONS, PRODUCT_MANAGEMENT } from '../constants/nav-items'
import { UserDropdown } from '../user-dropdown';
import { LoginButton } from '../login-button';

export class RigNav extends Component {
  openConfigurationsHandler = () => {
    this.props.openConfigurationsHandler();
  }

  render() {
    if (this.props.error !== '') {
      return (
        <div className="top-nav-error">
          <a> {this.props.error} </a>
        </div>
      );
    } else {
      return (
        <div className="top-nav">
          <a
            className={this.props.selectedView === EXTENSION_VIEWS ? "top-nav-item top-nav-item__selected" : "top-nav-item"}
            onClick={this.props.viewerHandler}>Extension Views</a>
          <a
            className={this.props.selectedView === BROADCASTER_CONFIG ? "top-nav-item top-nav-item__selected" : "top-nav-item"}
            onClick={this.props.configHandler}>Broadcaster Config</a>
          <a
            className={this.props.selectedView === LIVE_CONFIG ? "top-nav-item top-nav-item__selected" : "top-nav-item"}
            onClick={this.props.liveConfigHandler}>Live Config</a>
          <a
            className={this.props.selectedView === CONFIGURATIONS ? "top-nav-item top-nav-item__selected" : "top-nav-item"}
            onClick={this.openConfigurationsHandler}>Configurations</a>
          {!this.props.bitsEnabled && 
            <a
              className={this.props.selectedView === PRODUCT_MANAGEMENT ? "top-nav-item top-nav-item__selected" : "top-nav-item"}
              onClick={this.props.openProductManagementHandler}>Manage Products</a>
          }
          {this.props.login.login ? <UserDropdown login={this.props.login} /> : <LoginButton loginHandler={this.props.loginHandler}/>}
        </div>
      );
    }
  }
}

RigNav.propTypes = {
  openProductManagementHandler: PropTypes.func,
  openConfigurationsHandler: PropTypes.func,
  viewerHandler: PropTypes.func,
  configHandler: PropTypes.func,
  liveConfigHandler: PropTypes.func,
  selectedView: PropTypes.string,
  error: PropTypes.string,
  login: PropTypes.object,
};
