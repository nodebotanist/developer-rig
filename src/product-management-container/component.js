import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';
import { ProductTable } from '../product-table-component';

export class ProductManagementViewContainer extends Component {
  render() {
    return (
      <div className='view-container-wrapper'>
        <div className='view-container'>
          <ProductTable clientId={this.props.clientId} token={this.props.token} />
        </div>
      </div>
    );
  }
}

ProductManagementViewContainer.propTypes = {
  clientId: PropTypes.string.isRequired,
  token: PropTypes.string.isRequired
}