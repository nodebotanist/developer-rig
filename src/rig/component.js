import React, { Component } from 'react';
import { RigNav } from '../rig-nav';
import { ExtensionViewContainer } from '../extension-view-container';
import { ExtensionRigConsole } from '../console';
import { ExtensionViewDialog } from '../extension-view-dialog';
import { RigConfigurationsDialog } from '../rig-configurations-dialog';
import { EditViewDialog } from '../edit-view-dialog';
import { createExtensionObject } from '../util/extension';
import { createSignedToken } from '../util/token';
import { fetchManifest, fetchExtensionManifest, fetchUserInfo } from '../util/api';
import { EXTENSION_VIEWS, BROADCASTER_CONFIG, LIVE_CONFIG, CONFIGURATIONS, PRODUCT_MANAGEMENT } from '../constants/nav-items'
import { ViewerTypes } from '../constants/viewer-types';
import { OverlaySizes } from '../constants/overlay-sizes';
import { IdentityOptions } from '../constants/identity-options';
import { MobileSizes } from '../constants/mobile';
import { RIG_ROLE } from '../constants/rig';
import { ProductManagementViewContainer } from '../product-management-container/component';
const { ExtensionMode, ExtensionViewType } = window['extension-coordinator'];

export class Rig extends Component {
  constructor(props) {
    super(props);

    this.state = {
      clientId: process.env.EXT_CLIENT_ID,
      secret: process.env.EXT_SECRET,
      version: process.env.EXT_VERSION,
      channelId: process.env.EXT_CHANNEL_ID,
      userName: process.env.EXT_USER_NAME,
      mode: ExtensionMode.Viewer,
      extensionViews: [],
      manifest: {},
      showExtensionsView: false,
      showConfigurations: false,
      showEditView: false,
      showLoginView: false,
      showProductManagementView: false,
      idToEdit: 0,
      selectedView: EXTENSION_VIEWS,
      extension: {},
      userId: '',
      error: '',
      login: {},
    };
  }

  componentDidMount() {
    this._fetchInitialConfiguration();
  }

  componentWillMount() {
    this._initLocalStorage();
    this._setLogin();
  }

  openConfigurationsHandler = () => {
    this.setState({
      showConfigurations: true,
      selectedView: CONFIGURATIONS
    });
  }
  closeConfigurationsHandler = () => {
    this.setState({
      showConfigurations: false,
    });
  }

  openEditViewHandler = (id) => {
    this.setState({
      showEditView: true,
      idToEdit: id,
    });
  }

  closeEditViewHandler = () => {
    this.setState({
      showEditView: false,
      idToEdit: '0',
    });
  }

  viewerHandler = () => {
    this.setState({
      mode: ExtensionMode.Viewer,
      selectedView: EXTENSION_VIEWS,
      extension: {},
    });
  }

  configHandler = () => {
    this.setState({
      mode: ExtensionMode.Config,
      selectedView: BROADCASTER_CONFIG,
      extension: createExtensionObject(
        this.state.manifest,
        0,
        ViewerTypes.Broadcaster,
        '',
        this.state.userName,
        this.state.channelId,
        this.state.secret),
    });
  }

  liveConfigHandler = () => {
    this.setState({
      mode: ExtensionMode.Dashboard,
      selectedView: LIVE_CONFIG,
      extension: createExtensionObject(
        this.state.manifest,
        0,
        ViewerTypes.Broadcaster,
        '',
        this.state.userName,
        this.state.channelId,
        this.state.secret),
    });
  }

  openProductManagementHandler = () => {
    this.setState({
      selectedView: PRODUCT_MANAGEMENT,
    });
  }

  openExtensionViewHandler = () => {
    if (this.state.error === '') {
      this.setState({
        showExtensionsView: true,
      });
    }
  }

  closeExtensionViewDialog = () => {
    this.setState({
      showExtensionsView: false
    });
  }

  refreshConfigurationsHandler = () => {
    const token = createSignedToken(RIG_ROLE, '', this.state.userId, this.state.channelId, this.state.secret);
    fetchExtensionManifest('api.twitch.tv', this.state.clientId, this.state.version, token, this._onConfigurationSuccess, this._onConfigurationError);
  }

  loginHandler = () => {
    this.setState({
      showLoginView: true
    });
  }

  _onConfigurationSuccess = (data) => {
    this.setState(data);
  }

  _onConfigurationError = (errMsg) => {
    this.setState({
      error: errMsg,
    });
  }

  _getFrameSizeFromDialog(dialogRef) {
    if (dialogRef.state.frameSize === 'Custom') {
      return {
        width: dialogRef.state.width,
        height: dialogRef.state.height
      };
    }
    if (dialogRef.state.extensionViewType === ExtensionViewType.Mobile) {
      return MobileSizes[dialogRef.state.frameSize];
    }

    return OverlaySizes[dialogRef.state.frameSize];
  }

  createExtensionView = () => {
    const extensionViews = this._getExtensionViews();
    const linked = this.refs.extensionViewDialog.state.identityOption === IdentityOptions.Linked;
    extensionViews.push({
      id: (extensionViews.length + 1).toString(),
      type: this.refs.extensionViewDialog.state.extensionViewType,
      extension: createExtensionObject(
        this.state.manifest,
        (extensionViews.length + 1).toString(),
        this.refs.extensionViewDialog.state.viewerType,
        linked,
        this.state.userName,
        this.state.channelId,
        this.state.secret,
        this.refs.extensionViewDialog.state.opaqueId,
      ),
      linked: linked,
      role: this.refs.extensionViewDialog.state.viewerType,
      x: this.refs.extensionViewDialog.state.x,
      y: this.refs.extensionViewDialog.state.y,
      orientation: this.refs.extensionViewDialog.state.orientation,
      frameSize: this._getFrameSizeFromDialog(this.refs.extensionViewDialog),
    });
    this._pushExtensionViews(extensionViews);
    this.closeExtensionViewDialog();
  }

  deleteExtensionView = (id) => {
    this._pushExtensionViews(this.state.extensionViews.filter(element => element.id !== id));
  }

  editViewHandler = (newViewState) => {
    const views = this._getExtensionViews();
    views.forEach(element => {
      if (element.id === this.state.idToEdit) {
        element.x = newViewState.x;
        element.y = newViewState.y;
        element.orientation = newViewState.orientation;
      }
    });
    this._pushExtensionViews(views);
    this.closeEditViewHandler();
  }

  render() {
    let view = (
      <div>
        <ExtensionViewContainer
          ref="extensionViewContainer"
          mode={this.state.mode}
          extensionViews={this.state.extensionViews}
          deleteExtensionViewHandler={this.deleteExtensionView}
          openExtensionViewHandler={this.openExtensionViewHandler}
          openEditViewHandler={this.openEditViewHandler}
          extension={this.state.extension} />
        {this.state.showExtensionsView &&
          <ExtensionViewDialog
            ref="extensionViewDialog"
            extensionViews={this.state.manifest.views}
            show={this.state.showExtensionsView}
            closeHandler={this.closeExtensionViewDialog}
            saveHandler={this.createExtensionView} />}
        {this.state.showEditView &&
          <EditViewDialog
            ref="editViewDialog"
            idToEdit={this.state.idToEdit}
            show={this.state.showEditView}
            views={this._getExtensionViews()}
            closeHandler={this.closeEditViewHandler}
            saveViewHandler={this.editViewHandler}
          />}
        <RigConfigurationsDialog
          show={this.state.showConfigurations}
          config={this.state.manifest}
          closeConfigurationsHandler={this.closeConfigurationsHandler}
          refreshConfigurationsHandler={this.refreshConfigurationsHandler} />
        <ExtensionRigConsole />
      </div>
    );

    if (this.state.selectedView === PRODUCT_MANAGEMENT) {
      view = (
        <ProductManagementViewContainer clientId={this.state.clientId} token={this.state.accessToken} />
      );
    }

    return (
      <div>
        <RigNav
          ref="rigNav"
          login={this.state.login}
          bitsEnabled={this.state.manifest.bits_enabled}
          selectedView={this.state.selectedView}
          viewerHandler={this.viewerHandler}
          configHandler={this.configHandler}
          liveConfigHandler={this.liveConfigHandler}
          openConfigurationsHandler={this.openConfigurationsHandler}
          openProductManagementHandler={this.openProductManagementHandler}
          loginHandler={this.loginHandler}
          error={this.state.error}/>
        {view}
      </div>
    );
  }

  _getExtensionViews() {
    const extensionViewsValue = localStorage.getItem("extensionViews");
    return extensionViewsValue ? JSON.parse(extensionViewsValue) : extensionViewsValue;
  }

  _pushExtensionViews(newViews) {
    localStorage.setItem("extensionViews", JSON.stringify(newViews));
    this.setState({
      extensionViews: newViews,
    });
  }

  _fetchInitialConfiguration() {
    fetchManifest("api.twitch.tv", this.state.clientId, this.state.userName, this.state.version, this.state.channelId, this.state.secret, this._onConfigurationSuccess, this._onConfigurationError);
  }

  _initLocalStorage() {
    const extensionViewsValue = localStorage.getItem("extensionViews");
    if (!extensionViewsValue) {
      localStorage.setItem("extensionViews", JSON.stringify([]));
      return;
    }
    this.setState({
      extensionViews: JSON.parse(extensionViewsValue)
    })
  }

  _setLogin() {
    const windowHash = window.location.hash;
    const rigLogin = localStorage.getItem('rigLogin');
    if (windowHash.includes('access_token')) {
      const accessTokenKey = 'access_token=';
      const accessTokenIndex = windowHash.indexOf(accessTokenKey);
      const ampersandIndex = windowHash.indexOf('&');
      const accessToken = windowHash.substring(accessTokenIndex + accessTokenKey.length, ampersandIndex);
      fetchUserInfo('api.twitch.tv', accessToken, resp => {
        this.setState({
          accessToken: accessToken,
          login: resp,
        });
        localStorage.setItem('rigLogin', JSON.stringify({
          login: resp,
          accessToken: accessToken,
        }));
      }, err => {
        this.setState({
          error: err,
        });
      })
    }
    else if (rigLogin) {
      const login = JSON.parse(rigLogin);
      this.setState({
        accessToken: login.accessToken,
        login: login.login,
      })
    }
  }
}
