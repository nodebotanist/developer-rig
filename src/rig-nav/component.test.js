import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { RigNav } from './component';
import { EXTENSION_VIEWS, BROADCASTER_CONFIG, LIVE_CONFIG, CONFIGURATIONS, PRODUCT_MANAGEMENT } from '../constants/nav-items';

describe('<RigNav />', () => {
  const setupShallow = setupShallowTest(RigNav, () => ({
    login: {},
    loginHandler: jest.fn(),
    bitsEnabled: true,
    openConfigurationsHandler: jest.fn(),
    openProductManagementHandler: jest.fn(),
    viewerHandler: jest.fn(),
    configHandler: jest.fn(),
    liveConfigHandler: jest.fn(),
    selectedView: EXTENSION_VIEWS,
    error: '',
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an error', () => {
    const { wrapper } = setupShallow({
      error: 'test error',
    });
    expect(wrapper.find('.top-nav-error').text().trim()).toBe('test error');
  });

  it('correctly handles clicks on each tab', () => {
    const { wrapper } = setupShallow();
    wrapper.find('a.top-nav-item').forEach(tab => {
      tab.simulate('click');
    });
    expect(wrapper.instance().props.viewerHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.configHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.liveConfigHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.openConfigurationsHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.openProductManagementHandler).toHaveBeenCalled();
  });

  it('correct css classes are set when things are selected', () => {
    const { wrapper } = setupShallow({
      selectedView: EXTENSION_VIEWS,
    });
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);

    wrapper.setProps({
      selectedView: BROADCASTER_CONFIG,
    });
    wrapper.update();
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);

    wrapper.setProps({
      selectedView: LIVE_CONFIG,
    });
    wrapper.update();
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);

    wrapper.setProps({
      selectedView: CONFIGURATIONS,
    });
    wrapper.update();
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);

    wrapper.setProps({
      selectedView: PRODUCT_MANAGEMENT,
    });
    wrapper.update();
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);
  });

  it('correctly hides the Product Management tab when extension is bits enabled', () => {
    const { wrapper } = setupShallow({
      bitsEnabled: false
    });
    expect(wrapper.findWhere(el => el.text() === 'Manage Products').exists()).toBe(false);
  })
});
