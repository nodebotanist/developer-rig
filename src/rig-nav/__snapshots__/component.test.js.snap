// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RigNav /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigNav
    bitsEnabled={true}
    configHandler={[Function]}
    error=""
    liveConfigHandler={[Function]}
    login={Object {}}
    loginHandler={[Function]}
    openConfigurationsHandler={[Function]}
    openProductManagementHandler={[Function]}
    selectedView="Extension Views"
    viewerHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="top-nav-item top-nav-item__selected"
          onClick={[Function]}
>
          Extension Views
</a>,
        <a
          className="top-nav-item"
          onClick={[Function]}
>
          Broadcaster Config
</a>,
        <a
          className="top-nav-item"
          onClick={[Function]}
>
          Live Config
</a>,
        <a
          className="top-nav-item"
          onClick={[Function]}
>
          Configurations
</a>,
        <a
          className="top-nav-item"
          onClick={[Function]}
>
          Manage Products
</a>,
        <LoginButton
          loginHandler={[Function]}
/>,
      ],
      "className": "top-nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Extension Views",
          "className": "top-nav-item top-nav-item__selected",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Extension Views",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Broadcaster Config",
          "className": "top-nav-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Broadcaster Config",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Live Config",
          "className": "top-nav-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Live Config",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Configurations",
          "className": "top-nav-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Configurations",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Manage Products",
          "className": "top-nav-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Manage Products",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "loginHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="top-nav-item top-nav-item__selected"
            onClick={[Function]}
>
            Extension Views
</a>,
          <a
            className="top-nav-item"
            onClick={[Function]}
>
            Broadcaster Config
</a>,
          <a
            className="top-nav-item"
            onClick={[Function]}
>
            Live Config
</a>,
          <a
            className="top-nav-item"
            onClick={[Function]}
>
            Configurations
</a>,
          <a
            className="top-nav-item"
            onClick={[Function]}
>
            Manage Products
</a>,
          <LoginButton
            loginHandler={[Function]}
/>,
        ],
        "className": "top-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Extension Views",
            "className": "top-nav-item top-nav-item__selected",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Extension Views",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Broadcaster Config",
            "className": "top-nav-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Broadcaster Config",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Live Config",
            "className": "top-nav-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Live Config",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Configurations",
            "className": "top-nav-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Configurations",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Manage Products",
            "className": "top-nav-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Manage Products",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "loginHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
